backend/src/main/java/com/benine/backend/http/PresetCreationHandler.java

```diff
@@ -0,0 +1,108 @@
+import com.benine.backend.camera.Camera;
+import com.benine.backend.camera.CameraConnectionException;
+import com.benine.backend.camera.CameraController;
+import com.benine.backend.camera.ipcameracontrol.IPCamera;
+import com.benine.backend.database.DatabasePreset;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.jar.Attributes;
+import com.sun.net.httpserver.HttpExchange;
+
+/**
+ * Class allows creation of a preset by tagging a camera viewpoint location.
+**/
+public class PresetCreationHandler  extends RequestHandler {
+  
+ 
+  /**
+   * Create a new handler for creating new presets.
+   * @param controller the controller to interact with.
+   */
+  public PresetCreationHandler(CameraController controller) {
+    super(controller);
+  }
+  
+  /**
+     * Handles a request of making a new preset. 
+     * @param exchange the exchange containing data about the request.
+     * @throws IOException when an error occurs with responding to the request.
+  */
+  public void handle(HttpExchange exchange) throws IOException {
+    Attributes parsedURI;
+    try {
+      parsedURI = parseURI(exchange.getRequestURI().getQuery());
+        
+      int cameraID = Integer.parseInt(parsedURI.getValue("id"));
+      Camera camera =  getCameraController().getCameraById(cameraID);
+      IPCamera ipCamera = null;
+      
+      if (camera instanceof IPCamera) {
+        ipCamera = (IPCamera)camera;
+        
+        DatabasePreset preset = getCameraPositions(ipCamera);
+         
+        if (preset != null) {
+          //Create a random integer for the preset number, should later be changed.
+          Random randomGenerator = new Random();
+          int randomInt = randomGenerator.nextInt(100);
+          //Adding the new preset to the database
+          Main.getDatabase().addPreset(cameraID, randomInt, preset);
+          responseMessage(exchange, true);
+        } else {
+          responseMessage(exchange, false);
+        }
+      }
+    } catch (MalformedURIException e) {
+      responseMessage(exchange, false);
+      Main.getLogger().log("Wrong URI", LogEvent.Type.CRITICAL);;
+      return;
+    }   
+    
+  }
+  
+  /**
+   * @param ipCamera the ipCamera you want to get the position of.
+   * @return DatabasePreset preset.
+   * @throws CameraConnectionException thrown if camera isn't an IP Camera. 
+   */
+  public DatabasePreset getCameraPositions(IPCamera ipCamera) {
+    try {
+      //Get everything that is needed to create a new preset.  
+      int zoom = ipCamera.getZoomPosition();
+      int pan = (int)ipCamera.getPosition().getPan();
+      int tilt = (int)ipCamera.getPosition().getTilt();
+      int focus = ipCamera.getFocusPos();
+      int iris = ipCamera.getIrisPos();
+      boolean autofocus = ipCamera.isAutoFocusOn();
+    
+      //Create new DatabasePreset and return it.
+      DatabasePreset preset = new DatabasePreset(pan,tilt,zoom,focus,iris,autofocus);
+      return preset;
```

Comment:
You can return new `DatabasePreset(...)` immediately, without an extra variable.